### PR TITLE
chore: add more debug information for sync #WPB-11603

### PR DESCRIPTION
**Pull request by damian-kaczmarek**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11603" title="WPB-11603" target="_blank">BugWPB-11603</a>  [Android] App stuck on "decrypting messages"
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

Wasn't able to reproduce the problem. Added some more debug information. Possibly we should improve the UI to inform user about networking problems. Or listen for our requests failing not being respond for a long time.

### Dependencies (Optional)

more related changes in kalium: https://github.com/wireapp/wire-android/pull/3523

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/network/NetworkStateObserverImpl.kt [hunk 1]
```diff
@@ -76,14 +78,46 @@ internal actual class NetworkStateObserverImpl(
...
+                networkCapabilities: NetworkCapabilities
+            ) {
                 super.onCapabilitiesChanged(network, networkCapabilities)
-                kaliumLogger.i(
-                    "${NetworkStateObserver.TAG} capabilities changed " +
-                            "internet:${networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)} " +
-                            "validated:${networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)}"
-                )
-                defaultNetworkDataStateFlow.update { DefaultNetworkData.Connected(network, networkCapabilities) }
+                val loggerMessage = buildString {
```
Comment (yamilmedina) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/network/NetworkStateObserverImpl.kt:86]: Suggestion: We can try to log in `json` format, so we can later parse it and do some filtering in Datadog

You can take a look at this utility:

https://github.com/wireapp/kalium/blob/a2f0eadaaa0e9790b97b5aa35892749eb3bfb879/logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreLogger.kt#L63

**Review by yamilmedina — COMMENTED** [review]
Looks good ! Left a comment about datadog format that we can later use to filter better in the tool 🙌

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/network/NetworkStateObserverImpl.kt (hunk 1 above)
Comment (damian-kaczmarek) [reply logic/src/androidMain/kotlin/com/wire/kalium/logic/network/NetworkStateObserverImpl.kt]: thanks, added. Good to know we such method

**Review by yamilmedina — APPROVED** [review]
🚀

**Review by borichellow — APPROVED** [review]

**Pull request merged**